frontends/election-manager/src/components/print_button.tsx

```diff
@@ -1,151 +1,106 @@
+  );
 }
 
 interface PrintButtonProps extends StyledButtonProps {
-  title?: string;
-  afterPrint?: () => void;
-  afterPrintError?: (errorMessage: string) => void;
-  copies?: number;
-  sides: PrintOptions['sides'];
-  confirmModal?: ConfirmModal;
-  printTarget?: JSX.Element;
-  printTargetTestId?: string;
+  print: () => Promise<void>;
+  useDefaultProgressModal?: boolean;
 }
 
 export function PrintButton({
-  title,
-  afterPrint,
-  afterPrintError,
+  print,
+  useDefaultProgressModal = true,
   children,
-  copies,
-  sides,
-  confirmModal,
-  printTarget,
-  printTargetTestId,
   ...rest
 }: React.PropsWithChildren<PrintButtonProps>): JSX.Element {
-  const isMounted = useMountedState();
   const makeCancelable = useCancelablePromise();
-  const { printer } = useContext(AppContext);
-  const [isConfirming, setIsConfirming] = useState(false);
+  const { hasPrinterAttached } = useContext(AppContext);
+  const [isShowingConnectPrinterModal, setIsShowingConnectPrinterModal] =
+    useState(false);
   const [isPrinting, setIsPrinting] = useState(false);
-  const [showPrintingError, setShowPrintingError] = useState(false);
-  const [shouldRenderPrintTarget, setShouldRenderPrintTarget] = useState(false);
 
-  const print = useCallback(async () => {
-    if (window.kiosk) {
-      const printers = await makeCancelable(window.kiosk.getPrinterInfo());
-      if (!printers.some((p) => p.connected)) {
-        setShowPrintingError(true);
-        afterPrintError?.('No printer connected.');
-        return;
-      }
-    }
+  const needsPrinter = Boolean(window.kiosk) && !hasPrinterAttached;
 
+  const handlePrint = useCallback(async () => {
+    setIsShowingConnectPrinterModal(false);
     setIsPrinting(true);
-    setTimeout(() => {
-      if (!isMounted()) return;
-      setIsPrinting(false);
-    }, 3000);
-    const documentTitle = document.title;
-    if (title) {
-      document.title = title;
-    }
-    await makeCancelable(printer.print({ sides, copies }));
-    if (title) {
-      document.title = documentTitle;
-    }
-
-    setShouldRenderPrintTarget(false);
-    afterPrint?.();
-  }, [
-    afterPrint,
-    afterPrintError,
-    copies,
-    isMounted,
-    makeCancelable,
-    printer,
-    sides,
-    title,
-  ]);
-
-  useEffect(() => {
-    if (shouldRenderPrintTarget) {
-      void print();
-    }
-  }, [shouldRenderPrintTarget, print]);
-
-  const startPrint = useCallback(async () => {
-    if (printTarget) {
-      setShouldRenderPrintTarget(true);
-    } else {
-      await print();
-    }
-  }, [print, printTarget]);
-
-  function donePrintingError() {
-    setShowPrintingError(false);
-  }
-
-  function initConfirmModal() {
-    setIsConfirming(true);
-  }
-
-  function cancelPrint() {
-    setIsConfirming(false);
-  }
-
-  async function confirmPrint() {
-    setIsConfirming(false);
     await print();
-  }
+    if (useDefaultProgressModal) {
+      await makeCancelable(sleep(DEFAULT_PROGRESS_MODAL_DELAY_SECONDS * 1000));
```

Comment:
I think this could result in a slight change from the previous behavior if `await print()` takes some time to complete. Not sure it matters though. (Previously we started the delay before starting printing)

Reply:
Yeah. If we do start basing timing off of printer feedback, then this will change again.